### PR TITLE
kernel: modules/lib-lz4: add lz4hc_compress

### PR DESCRIPTION
**Pull request by Spudz76**
> enable option `CONFIG_CRYPTO_LZ4HC` to match default kernel config
> 
> this only adds the `lz4hc_compress` module, and has no effect on the `lz4_decompress` module which already supports any flavor
   
The resulting size of the `lz4hc_compress.ko` on a `mipsel` (`mt7621`) target for kernel `5.15.80` is `11544` bytes and uses `10304` bytes in memory while loaded.

This compressor has been supported by at least ubifs for a long time, so this may have a curative side effect on making ubifs-with-lz4hc work properly but that has not been tested.  It may currently silently fall back to lz4_compress which will still work, but does not match expected/assumed behavior.  Reading from one as read-only (ubifs-root) already works because either format can be unpacked by a single decompressor, so bugs may not be noticed, and probably not many users are trying to use lz4hc ubifs.

### CODE REVIEW

**Comment by Spudz76** [comment]
Also, if the lz4 modules end up built-in, they would include this module, and then everything would work.  So, anyone using ubifs-root with lz4hc would be working properly already.  The curative may only be for those (0 users) that would use add-on packages to make another storage region somewhere on-device.  When the lz4 modules are built-in the kmod loader throws errors at boot, but does not seem to harm anything and all other expected modules end up loaded.  Somewhere else, I suspect in the configuration tree, something should disable installation of this package if the modules are built-in and that would cure the innocuous but possibly confusing warning-spam (but I have not investigated into that yet).

**Comment by Spudz76** [comment]
I believe all the build failures are unrelated to this, the several I checked were hung up on an `oldconfig` phase in the ZRAM section, as if there is no default to auto-select, so it sits there awaiting input from nobody until timeout.

**Comment by Spudz76** [comment]
no changes, just rebased on latest master (and maybe trigger the checks to pass since they crashed on nothing related last time)

**Comment by hauke** [comment]
The problems from the CI are related to your changes.

It found this problem:
````
Run make target/compile -j$(nproc) BUILD_LOG=1 || ret=$? .github/workflows/scripts/show_build_failures.sh
make[2]: Entering directory '/__w/openwrt/openwrt/openwrt/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/__w/openwrt/openwrt/openwrt/scripts/config'
 make[1] target/compile
 make[2] -C target/linux compile
    ERROR: target/linux failed to build.
make -r target/compile: build failed. Please re-run make with -j1 V=s or V=sc for a higher verbosity level to see what's going on
make: *** [/__w/openwrt/openwrt/openwrt/include/toplevel.mk:231: target/compile] Error 1
====== Make errors from logs/target/linux/compile.txt ======
  Normal floppy disk support (BLK_DEV_FD) [N/m/y/?] n
  Parallel port IDE device support (PARIDE) [N/m/?] n
  Block Device Driver for Micron PCIe SSDs (BLK_DEV_PCIESSD_MTIP32XX) [N/m/y/?] n
  Compressed RAM block device support (ZRAM) [M/n/?] m
    Default zram compressor
    > 1. lzo-rle (ZRAM_DEF_COMP_LZORLE)
      2. zstd (ZRAM_DEF_COMP_ZSTD)
      3. lz4 (ZRAM_DEF_COMP_LZ4)
      4. lzo (ZRAM_DEF_COMP_LZO)
      5. lz4hc (ZRAM_DEF_COMP_LZ4HC) (NEW)
    choice[1-5?]: make[6]: *** [scripts/kconfig/Makefile:77: syncconfig] Error 1
make[5]: *** [Makefile:620: syncconfig] Error 2
make[4]: *** [Makefile:740: include/config/auto.conf.cmd] Error 2
make[3]: *** [Makefile:26: /__w/openwrt/openwrt/openwrt/build_dir/target-powerpc-openwrt-linux-musl_musl/linux-apm821xx_nand/linux-5.15.80/.modules] Error 2
make[2]: *** [Makefile:11: compile] Error 2
time: target/linux/compile#26.55#9.52#32.22
```

When using the following configuration:
````
CONFIG_TARGET_apm821xx=y
CONFIG_TARGET_apm821xx_nand=y
CONFIG_TARGET_apm821xx_nand_Default=y
CONFIG_ALL_KMODS=y
````

**Comment by Spudz76** [comment]
But I didn't touch any ZRAM anything?  If it was any of those it would be `ZRAM_DEF_COMP_LZ4HC` because that's the compressor I enabled, which probably made that option available (thus "NEW")

File: target/linux/generic/config-5.10 [hunk 1]
```diff
@@ -7184,6 +7184,11 @@ CONFIG_ZONE_DMA=y
 # CONFIG_ZPA2326 is not set
 # CONFIG_ZPOOL is not set
 # CONFIG_ZRAM is not set
+# CONFIG_ZRAM_DEF_COMP_LZ4 is not set
+# CONFIG_ZRAM_DEF_COMP_LZ4HC is not set
+# CONFIG_ZRAM_DEF_COMP_LZO is not set
+# CONFIG_ZRAM_DEF_COMP_LZORLE is not set
+# CONFIG_ZRAM_DEF_COMP_ZSTD is not set
```
Comment (hauke) [inline target/linux/generic/config-5.10:7191]: kernel 5.10 does not have these options, they were added with kernel 5.11.

Comment (Spudz76) [reply target/linux/generic/config-5.10:7191]: Good, then the Kconfig phase will simply remove them.

**Comment by Spudz76** [comment]
Split out that 842 into the PR that adds the modules so it's just all done.

**Comment by hauke** [comment]
You second commit says 
> this will be smashed into one commit if it works
When do you plan to do this?

**Comment by Spudz76** [comment]
Oh good, now it doesn't pass anymore, even though the only difference is smashed commits.  neat.

**Comment by hauke** [comment]
Please rebase again, there were some problems in master which are hopefully fixed now.

**Pull request merged**